moz/l10n/paths/discover.py

```diff
@@ -78,9 +58,31 @@ def __init__(
         self,
         root: str,
         ref_root: str | None = None,
+        *,
         ignorepath: str | None = ".l10n-ignore",
+        source_locale: str | list[str] | None = None,
     ) -> None:
         """
+        Automagical localization resource discovery.
+
+        Given a `root` directory, finds the likeliest reference and target directories.
+
+        The reference directory has a name matching the `source_locale`
+        and contains files with extensions that appear localizable.
+
+        If `ref_root` is given, the reference directory must be within it.
+        If it does not contain a directory matching `source_locale`
+        but `ref_root` itself contains files with extensions that appear localizable,
+        it is used as the reference directory.
+
+        The default `source_locale` is `['en-US', 'en']`,
```

Comment:
What happens with "templates"?

Reply:
Not treated as anything special. From Pontoon we [call this](https://github.com/mozilla/pontoon/pull/3312/files#diff-b18de80c1f885538e633fd3935679aa1db868c06afc4eecf9af1341df7b707aaR29) with `ref_root` set to the source repo root, which doesn't need to have any special name in order to fall back to using the repo root as the reference root.

Reply:
How is it going to detect that https://github.com/mozilla/addons-server/tree/master/locale/templates is the source locale folder, and not https://github.com/mozilla/addons-server/tree/master/locale/en_US?

Reply:
At least for now, we'll need to call this from Pontoon with `source_locale=["templates", "en-US", "en"]` to keep its behaviour consistent.

Reply:
Meh, I completely missed `ref_paths` was a function parameter.